src/metatrain/utils/additive/zbl.py

```diff
@@ -0,0 +1,309 @@
+            if target.quantity != "energy":
+                raise ValueError(
+                    "ZBL only supports energy-like outputs, but a "
+                    f"{target.quantity} output was provided."
+                )
+            if target.unit != "eV":
+                raise ValueError(
+                    "ZBL only supports eV units, but a "
+                    f"{target.unit} output was provided."
+                )
+
+        self.dataset_info = dataset_info
+        self.atomic_types = sorted(dataset_info.atomic_types)
+
+        n_types = len(self.atomic_types)
+
+        self.output_to_output_index = {
+            target: i for i, target in enumerate(sorted(dataset_info.targets.keys()))
+        }
+
+        self.register_buffer(
+            "species_to_index",
+            torch.full((max(self.atomic_types) + 1,), -1, dtype=torch.int),
+        )
+        for i, t in enumerate(self.atomic_types):
+            self.species_to_index[t] = i
+
+        self.register_buffer(
+            "covalent_radii", torch.empty((n_types,), dtype=torch.float64)
+        )
+        for i, t in enumerate(self.atomic_types):
+            self.covalent_radii[i] = covalent_radii[t]
+
+        self.largest_covalent_radius = float(torch.max(self.covalent_radii))
+
+    def restart(self, dataset_info: DatasetInfo) -> "ZBL":
+        """Restart the model with a new dataset info.
+
+        :param dataset_info: New dataset information to be used.
+        """
+        return self({}, self.dataset_info.union(dataset_info))
+
+    def forward(
+        self,
+        systems: List[System],
+        outputs: Dict[str, ModelOutput],
+        selected_atoms: Optional[Labels] = None,
+    ) -> Dict[str, TensorMap]:
+        """Compute the energies of a system solely based on a ZBL repulsive
+        potential.
+
+        :param systems: List of systems to calculate the ZBL energy.
+        :param outputs: Dictionary containing the model outputs.
+        :param selected_atoms: Optional selection of atoms for which to compute the
+            predictions.
+        :returns: A dictionary with the computed predictions for each system.
+
+        :raises ValueError: If the `outputs` contain unsupported keys.
+        """
+
+        # Assert only one neighbor list for all systems
+        neighbor_lists: List[TensorBlock] = []
+        for system in systems:
+            nl_options = system.known_neighbor_lists()
+            if len(nl_options) != 1:
+                raise ValueError("ZBL only supports one neighbor list per system.")
+            nl_option = nl_options[0]
+            if nl_option.cutoff < 2.0 * self.largest_covalent_radius:
+                raise ValueError(
+                    "ZBL only supports neighbor lists with a cutoff of at least "
+                    f"{2.0 * self.largest_covalent_radius} Å, since the largest "
+                    f"covalent radius is {self.largest_covalent_radius} Å."
+                )
```

Comment:
Why?

Reply:
This is needed to make the potential smooth. If the ZBL outer cutoff is larger than the NL cutoff, this will lead to a discontinuity in the ZBL potential

Reply:
I should mention that this is an extremely unlikely scenario if people use sensible cutoffs (4-5 A), even for big atoms. @HannaTuerk

Reply:
Yeah, makes sense. Maybe just give the `cutoff` again here?

Reply:
So if ZBL requires it's own neighbor list, it can request a separate NL from the rest of the model. The engine will then provide multiple NL (one for ZBL, one or more for the model), and everyone is happy.